Source: http://developer.pidgin.im/viewmtn/revision/info/bb09c1ec048f9cf29c1b8de94c5781a39878380e
Upstream: yes, http://developer.pidgin.im/ticket/10966#comment:2
Reason: libtool tries to include the system libpurple instead, leads to
segfaulting tests.

#
#
# patch "libpurple/tests/Makefile.am"
#  from [e099af98ceb831b19c8096503a3cb39d433456fb]
#    to [1f439ae5170a6f68f80997749d5dc71df844fe11]
#
============================================================
--- libpurple/tests/Makefile.am	e099af98ceb831b19c8096503a3cb39d433456fb
+++ libpurple/tests/Makefile.am	1f439ae5170a6f68f80997749d5dc71df844fe11
@@ -27,11 +27,11 @@ check_libpurple_LDADD=\
 		-DBUILDDIR=\"$(top_builddir)\"
 
 check_libpurple_LDADD=\
-        @CHECK_LIBS@ \
-		$(GLIB_LIBS) \
 		$(top_builddir)/libpurple/protocols/jabber/libjabber.la \
 		$(top_builddir)/libpurple/protocols/qq/libqq.la \
 		$(top_builddir)/libpurple/protocols/yahoo/libymsg.la \
-		$(top_builddir)/libpurple/libpurple.la
+		$(top_builddir)/libpurple/libpurple.la \
+        @CHECK_LIBS@ \
+		$(GLIB_LIBS)
 
 endif
